charts/vald/values.yaml

```
@@ -1069,6 +1069,11 @@ gateway:
       # @schema {"name": "gateway.lb.ingress.servicePort", "type": "string"}
       # gateway.lb.ingress.servicePort -- service port to be exposed by ingress
       servicePort: grpc
+      # @schema {"name": "gateway.lb.ingress.defaultBackend", "type": "object"}
+      defaultBackend:
+        # @schema {"name": "gateway.lb.ingress.defaultBackend.enabled", "type": "boolean"}
+        # gateway.lb.ingress.defaultBackend.enabled -- gateway ingress defaultBackend enabled
+        enabled: true
```

It would be better to define the IngressBackend object.
Ref: https://kubernetes.io/docs/reference/kubernetes-api/service-resources/ingress-v1/#IngressBackend

e.g.
```suggestion
      # @schema {"name": "gateway.lb.ingress.defaultBackend", "type": "object"}
      defaultBackend:
        # @schema {"name": "gateway.lb.ingress.defaultBackend.enabled", "type": "boolean"}
        # gateway.lb.ingress.defaultBackend.enabled -- gateway ingress defaultBackend enabled
        enabled: true
        resource:
        ...
        service:
        ...
```

And also, it requires updating the helm template.

---

Thanks for the review.
Currently, Vald's defaultBackend.service is set to the same properties as the ingress backend.
https://github.com/vdaas/vald/blob/main/charts/vald/templates/gateway/filter/ing.yaml#L36-L37C1

If we were to allow fine-tuning of defaultBackend based on the Kubernets specification, there would be a difference between this behavior and the one described above.

If this is to be absorbed, a flag like `followIngressProperty` would need to be added to allow the defaultBackend to follow the ingress.backend as before, or to be configured on its own.

I'm thinking that it would be better to simply be able to control whether defaultBackend is enabled/disabled in this case.

---

> Currently, Vald's defaultBackend.service is set to the same properties as the ingress backend.
> https://github.com/vdaas/vald/blob/main/charts/vald/templates/gateway/filter/ing.yaml#L36-L37C1

It's correct in the current.
IMO, it is the chance to refactor templates to apply both IngressBackend configurations for available.
It will make help users can select which option to use.
So, I think applying the IngressBackend configuration object in this situation would be better.